docs: Add a Testmonials page.

PhilipMetzger opened the pull request:
Show our positive reviews to the world. It's also not completly serious, in a similar fashion to Phabricator.

[Rendered](https://github.com/martinvonz/jj/blob/push-rvuprqxytrtu/docs/testimonials.md)

cc @thoughtpolice

benbrittain commented on docs/testimonials.md:
```diff
@@ -0,0 +1,39 @@
+
+> Wait, it's not called Jujitsu?
+
+Phil, an Mercurial contributor and user.
```
an -> a

PhilipMetzger replied:
done.

benbrittain reviewed (approved):
I approve my testimonial 👍

thoughtpolice commented:
I went ahead and gave a slight rework of this page since I didn't want us to go through a bunch of review cycles for small things. Let me know what you think. It also adopts the "shill for us" joke from Phab. (Gist for solo reference: https://gist.github.com/thoughtpolice/68d1fc803e53e64571b7b71a9ef476e3)

---

# Testimonials

You might not be ready to make the jump to Jujutsu yet. That's OK &mdash; in order to provide some motivation, we've collected a number of valuable testimonials from both users and developers, which are sure to tip the scales and manipulate you in our favor!

## What users have to say

> Jujutsu is amazing... I couldn't have come up with anything remotely as elegant.
> 
> It's so rare that a solution attacks the innermost core of a problem so 
> thoroughly, I genuinely feel blessed to be in its presence. And also a bit 
> vindicated in not even trying to learn to use any of the tools that felt like
> more crutches stacked upon a sand castle

&mdash; Anonymous user

> It's the easiest time I've ever had learning a tool this deeply this quickly, 
> because of the ability to experiment and undo, instead of triple-checking 
> before trying a new scary command.

&mdash; Scott Olson

> Jujutsu is cool. And you can keep most of your existing workflows, too

&mdash; Ben Brittain, who changed his workflow seconds after saying this to his coworkers

> Wait, it's not called Jujitsu?

&mdash; Phil, Mercurial contributor

## What the developers have to say

> As a completely, totally, 100% unbiased source, I think it's pretty cool.

&mdash; Martin von Zweigbergk, creator of the project and completely biased source

> I'm sometimes still surprised that `jj next` and `jj prev` work

&mdash; Philip Metzger, contributor, author of `jj next` and `jj prev`

> "The phone number you have dialed is no longer in service. Please hang up and try a new number."

&mdash; Austin Seipp, contributor

# Spread the word yourself

Satisfied? Overwhelmed with joy? Ready to recommend Jujutsu to a Jujillion people? Good, because we have list of Pre-Approved Endorsements, designed in a laboratory by our PR team, ready to go. Start shilling for us immediately &mdash; and your check will be in the mail soon enough!

> Jujutsu seems alright.

> Jujutsu is my favorite software tool of all time. I say this for no particular reason.

> I love Jujutsu. I love Jujutsu. I love Jujutsu. I love Jujutsu. I love Jujutsu. I love Jujutsu. I love Jujutsu. I love Jujutsu.

benbrittain commented:
I'd rather not have it phrased like this for me, and I'd prefer to have my last name left off this page.

jyn514 commented:
how about this one?

> only breaks half as often as my other tools!
> — jyn, tool abuser

PhilipMetzger commented:
> how about this one?
> 
> > only breaks half as often as my other tools!
> > — jyn, tool abuser

May I add your ex-rustc hat to the footnote?

jyn514 commented:
you may! if you like you may also quote me as saying my job when i worked on rustc's bootstrap system was to chain together as many cursed combinations of tools as possible

necauqua commented:
> I made it so that commits have green checkmarks in the log when they do on GitHub, gives me a dopamine hit

Anton Bulakh, contributor, worked on cryptographic commit signing support that never landed into `main` (yet, hopefully)

---
sorry just got back from answering some stuff about that 😂

sunshowers commented:
Here's one from me:

I've spent many years of my career working on version control. What I like most about Jujutsu is how it has non-obvious solutions to UX problems that we've run into in the past. What most people may not realize is that there are many features in Jujutsu which interlock to make it easy to use.

For example, consider Jujutsu's support for automatically rebasing descendants of amended revisions. When we implemented that, we ran into an issue: what if there's a merge conflict? Our solution was to warn users and just not perform the auto-rebase. Now, suddenly, users have to understand that there can be old versions of the same revision visible in their log, and learn how to fix this state.

In contrast, Jujutsu's solution is to simply make merge conflicts first-class. This is not just an improvement in general, it is also _specifically_ an improvement for auto-rebase -- users no longer have to learn about old versions of a revision unless they want to look at the obslog.

Over and over, I'm struck by how well Jujutsu demonstrates this kind of evolved thinking, which as an experienced version control developer I deeply appreciate.

- Rain, engineer at Oxide and former VCS developer

poliorcetics commented:
Following the others:

I've started using `jj` for personal repos and it has quickly gone from "neat, let's try this more" to "*very* neat, added to my permanent config and automatically installed for new machines".

- Poliorcetics

PhilipMetzger commented on docs/testimonials.md:
```diff
@@ -0,0 +1,92 @@
+
+Austin Seipp, contributor.
+
+> As a fan of big green checkmarks in the log, made it so that Github is a ppure dopamine hit. 
+
+Anton Bulakh, contributor, author of the cryptographic commit signing support.
```
Does that look good @necauqua?

PhilipMetzger commented on docs/testimonials.md:
```diff
@@ -0,0 +1,92 @@
+
+An anonymous user.
+
+> only breaks half as often as my other tools!
+
+jyn, ex-rustc bootstrap developer, a place where tools break down all the time.
```
@jyn514, is this good?

phillco commented on docs/testimonials.md:
```diff
@@ -0,0 +1,92 @@
+
+> Jujutsu is my favorite software tool of all time. I say this for no particular reason.
+
+> I love Jujutsu. I love Jujutsu. I love Jujutsu. I love Jujutsu. I love Jujutsu. I love Jujutsu. I love Jujutsu. I love Jujutsu. 
```
Big fan of the "GPT hallucination vibes" with this one

phillco commented on docs/testimonials.md:
```diff
@@ -0,0 +1,92 @@
+
+Austin Seipp, contributor.
+
+> As a fan of big green checkmarks in the log, made it so that Github is a ppure dopamine hit. 
```
```suggestion
> As a fan of big green checkmarks in the log, jj made it so that Github is a pure dopamine hit. 
```

PhilipMetzger replied:
That's a great suggestion, thanks. I've only replace `jj` with Jujutsu.

PhilipMetzger commented on docs/testimonials.md:
```diff
@@ -0,0 +1,97 @@
+
+jyn, ex-rustc bootstrap developer, a place where tools break down all the time.
+
+> I initially started to use Jujutsu for personal repos where it has quickly 
+> gone from "neat, let's try this more" to "very neat, added to my permanent
+> config and automatically installed for new machines". 
+
+Poliorcetics
```
@poliorcetics, thanks for the testimonial, is this good for you?

poliorcetics replied:
English question (not a native speaker): is it better to say `personal repos where it has` or `personal repos and it has` ? I'm unsure about the differences and I used `where` by habit I think

jyn514 replied:
i would either say "personal repos, where" or "personal repos and". "personal repos where" without a comma looks odd to my eye.

sunshowers replied:
I think either gets the point across, though I would personally use `and it has`, and also add a comma:

`Jujutsu for personal repos, and it has`

But that's just my stylistic choice. It's completely fine as-is.

poliorcetics replied:
Then @PhilipMetzger `, and` or `, where` are fine :)

PhilipMetzger replied:
I've added the comma (`, and`) and reverted it to the fully spelled out version.

poliorcetics replied:
Yes thanks :)

PhilipMetzger replied:
No, thank you for your contribution :-).

sunshowers commented on docs/testimonials.md:
```diff
@@ -0,0 +1,97 @@
+> versions of a revision unless they want to look at the obslog.
+> 
+> Over and over, I'm struck by how well Jujutsu demonstrates this kind of 
+> evolved thinking, As an experienced version control developer, 
```
```suggestion
> evolved thinking. As an experienced version control developer, 
```

sunshowers replied:
(not your fault, I just caught it myself)

thoughtpolice commented:
I've done a big rework and added a bunch of quotes. There are a number of guiding ideas here behind some of the tweaks I've made.

- Some clarical and typographic fixes; for example, always use `&mdash;` to indicate the author after a block quote, so it looks closer to a typical typographic quote in a book. Also use the HTML enetity rather than the UTF-8 character, so that it's obvious when writing new quotes what to use. This avoids inconsistencies like em-dash versus hypen, and also solves the usage of `--` in some spots (fun fact: that's just a a typographic holdover from typewriters, when em dash was difficult or impossible to type! so you use hypen twice and slightly overlap them)

- The levity comes later. I've added all the more "serious" quotes first, gradually devolving a little more and more over time. Everyone does want to be a comedian, but of course some of us have to actually make the pitch. :) Otherwise it obviously far too "overdesigned" and bloated.

- I made some minor changes to some of the quotes, including

  - Reverting @benbrittain quote to it's original state. If Ben wants to add any more levity or change the punchline I'll do so!

  - Minor fixes for some run ons (e.g. `...` instead) from the anonymous quote; this is mostly just to make this particular one read more clearly.

  - I liked @jyn514's quote, but I wanted it to have a little more context. A joke is 99% setup, then the punchline. I also have a bunch of experience working on compilers and ridiculously fucked up tools. So I changed it to this:

    > when I worked on the rust compiler, my job was to chain together a bunch of strange and cursed tools that broke often. jujutsu breaks about half as much, so that's pretty good i guess

     I feel this retains the spirit of the quote, the kind of attitude that can only come from working on that stuff, where you now have a 1000 yard stare so your most loving endorsement of anything computer related is something like "it's alright". While also remaining in-character and with the right "voice". Please let me know if this is bad.

  - Very minor tweaks to Rain's testimonial, mostly to use the 1st person form less e.g. just less occurrences of the word "Jujutsu" in favor of implicit reference. Otherwise I put it up front and center.

  - I changed my own testimonial to a serious one, and mine comes first on the developer side; again this is so the sillier ones in this particular section come later. This is a huge blow to me, personally, as I don't get to be a class clown in this case. However, it is a cross I am willing to bear, all for you, my lovely subjects. If anyone else who writes patches wants to go "serious business mode" with me, that would be great.

  - I also liked @necauqua's quote but I wanted it to land a bit differently so I modified the tone a bit. 

    > Honestly, I implemented signing support mostly for that sweet dopamine hit that you get from the green checkmark on GitHub. Yeah.

    The idea was something more like, what would it sound like if you randomly went up to a guy on the street and asked a question?

  - The bottom section is completely rewritten. I think it's a bit better without the corpo associations.

Please let me know what you all think. I hope I'm not stepping on toes here!

sunshowers commented:
The changes work for me! Always appreciate a good copy-edit.

jyn514 commented:
> * I liked @jyn514's quote, but I wanted it to have a little more context. A joke is 99% setup, then the punchline. I also have a bunch of experience working on compilers and ridiculously fucked up tools. So I changed it to this:

> > * when I worked on the rust compiler, my job was to chain together a bunch of strange and cursed tools that broke often. jujutsu breaks about half as much, so that's pretty good i guess

hmm. i *love* the thousand yard stare idea but i want to make it more upbeat. how about this?

> when i worked on the rust compiler, my job was to chain together tools in the most cursed way possible. Jujutsu only breaks about half as often as those chains!

(and please do keep "i" lowercase)

---

> * I feel this retains the spirit of the quote, the kind of attitude that can only come from working on that stuff, where you now have a 1000 yard stare so your most loving endorsement of anything computer related is something like "it's alright".

LMAO you're so right though

sunshowers commented on docs/testimonials.md:
```diff
@@ -0,0 +1,118 @@
+>
+> In contrast, Jujutsu's solution is to simply make merge conflicts first-class.
+> This is not just an improvement in general, it is also specifically an
+> improvement for auto-rebase &mdash; users no longer have to learn about old
```
Interestingly, most style guides say that you [shouldn't put spaces around em-dashes](https://prowritingaid.com/grammar/1000066/Should-I-use-spaces-around-an-em-dash). Like many people I've been in the habit of putting spaces around em-dashes, though I've been trying to move away from that.

Either way's fine and consistency is most important, but just wanted to mention that.

phillco commented on docs/testimonials.md:
```diff
@@ -0,0 +1,118 @@
+
+&mdash; Anton Bulakh, contributor and dopamine enthusiast
+
+> I'm sometimes still surprised that navigating with `next` and `prev` works.
```
```suggestion
> I'm sometimes still surprised that navigating with `jj next` and `jj prev` works.
```

should probably be symmetric with the punchline

PhilipMetzger replied:
done

PhilipMetzger commented:
> > when i worked on the rust compiler, my job was to chain together tools in the most cursed way possible. Jujutsu only breaks about half as often as those chains!
> 
> (and please do keep "i" lowercase)

I've adjusted it back to your spelling.

gulbanana commented:
> When I heard about Jujutsu I decided to try it out before forming an opinion. Technically it never formed, because I haven't considered going back. 
- gul banana, computer programmer

PhilipMetzger commented:
> > When I heard about Jujutsu I decided to try it out before forming an opinion. Technically it never formed, because I haven't considered going back.
> 
> * gul banana, computer programmer

I've added your quote and reverted Martins for now.

PhilipMetzger commented:
I'm going to mark this as ready for review, as I'm planning to land this on next friday or saturday.
